lib/cocina/rspec/factories.rb

```diff
@@ -0,0 +1,155 @@
+      # rubocop:enable Metrics/ParameterLists
+
+      def self.build_dro(attributes)
+        Cocina::Models.build(build_dro_properties(**DRO_DEFAULTS.merge(attributes)))
+      end
+
+      # rubocop:disable Metrics/ParameterLists
+      def self.build_collection_properties(type:, id:, version:, label:, title:, admin_policy_id:, source_id: nil, catkeys: [])
+        {
+          type: type,
+          externalIdentifier: id,
+          version: version,
+          label: label,
+          access: {},
+          administrative: { hasAdminPolicy: admin_policy_id },
+          description: {
+            title: [{ value: title }],
+            purl: "https://purl.stanford.edu/#{id.delete_prefix('druid:')}"
+          },
+          identification: {}
+        }.tap do |props|
+          if catkeys.present?
+            props[:identification][:catalogLinks] = catkeys.map.with_index do |catkey, index|
+              { catalog: 'symphony', catalogRecordId: catkey, refresh: index.zero? }
+            end
+          end
+          props[:identification][:sourceId] = source_id if source_id
+        end
+      end
+      # rubocop:enable Metrics/ParameterLists
+
+      def self.build_collection(attributes)
+        Cocina::Models.build(build_collection_properties(**COLLECTION_DEFAULTS.merge(attributes)))
+      end
+
+      def self.build_admin_policy(attributes)
+        Cocina::Models.build(build_admin_policy_properties(**ADMIN_POLICY_DEFAULTS.merge(attributes)))
+      end
+
+      # rubocop:disable Metrics/ParameterLists
+      def self.build_admin_policy_properties(type:, id:, version:, label:, title:,
+                                             admin_policy_id:, agreement_id:,
+                                             use_statement: nil, copyright: nil, license: nil,
+                                             registration_workflow: nil, collections_for_registration: nil)
+        {
+          type: type,
+          externalIdentifier: id,
+          version: version,
+          label: label,
+          administrative: {
+            hasAdminPolicy: admin_policy_id,
+            hasAgreement: agreement_id,
+            accessTemplate: {
+              view: 'world',
+              download: 'world'
+            }
+          },
+          description: {
+            title: [{ value: title }],
+            purl: "https://purl.stanford.edu/#{id.delete_prefix('druid:')}"
+          }
+        }.tap do |props|
+          props[:administrative][:accessTemplate][:useAndReproductionStatement] = use_statement if use_statement
+          props[:administrative][:accessTemplate][:copyright] = copyright if copyright
+          props[:administrative][:accessTemplate][:license] = license if license
+          props[:administrative][:registrationWorkflow] = registration_workflow if registration_workflow
+          props[:administrative][:collectionsForRegistration] = collections_for_registration if collections_for_registration
+        end
+      end
```

Comment:
Would you consider also pulling in https://github.com/sul-dlss/argo/commit/14bae58303082e5b2efa0468355e765995af0df5#diff-221dfecc29dca1e150879df576b7900ec2e67879af2c64a3131abb52f886956d ?